### PR TITLE
chore(project): update to django 5.0

### LINKED ISSUE #9024: Update to Django 5.0
Django 4 is here and is very nice, let's update!

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-3632)

### PR DESCRIPTION
**Pull request by jaredlockhart**
Because

* Django 5.0 is out now
* We should stay up to date with the latest Django

This commit

* Updates to Django 5.0

fixes #9024

### CODE REVIEW

File: experimenter/experimenter/openidc/middleware.py [hunk 1]
```diff
@@ -28,7 +28,10 @@ def __call__(self, request):
             pass
 
         default_email = settings.DEV_USER_EMAIL if settings.DEBUG else None
-        openidc_email = request.META.get(settings.OPENIDC_EMAIL_HEADER, default_email)
+        openidc_email = request.META.get(
+            settings.OPENIDC_EMAIL_HEADER,
+            request.META.get(f"HTTP_{settings.OPENIDC_EMAIL_HEADER}", default_email),
```
Comment (brennie) [inline experimenter/experimenter/openidc/middleware.py:33]: Why this change in this patch?

File: experimenter/experimenter/legacy/legacy-ui/templates/experiments/list.html [hunk 2]
```diff
@@ -156,7 +156,7 @@ <h5>{{ experiment.population }}</h5>
           </li>
         {% endif %}
         {% for page_num in page_obj.paginator.page_range %}
-          <li class="page-item {% ifequal page_obj.number page_num %}active{% endifequal %}">
+          <li class="page-item {% if page_obj.number == page_num %}active{% endif %}">
```
Comment (brennie) [inline experimenter/experimenter/legacy/legacy-ui/templates/experiments/list.html:159]: ITS SO BEAUTIFUL 😭

File: experimenter/experimenter/openidc/middleware.py (hunk 1 above)
Comment (jaredlockhart) [reply experimenter/experimenter/openidc/middleware.py:33]: So something somewhere is rewriting headers to prepend `HTTP_` so now `HTTP_X_FORWARDED_USER` appears as `HTTP_HTTP_X_FORWARDED_USER` but only sometimes?  So I guess now we have to check for both 🤷‍♀️

Comment (jaredlockhart) [reply experimenter/experimenter/openidc/middleware.py:33]: Actually yeah I'll add a comment there about it.  It might be the test client, it's not clear.

File: experimenter/pyproject.toml [hunk 3]
```diff
@@ -73,6 +70,9 @@ django-redis = "^5.4.0"
 fontawesomefree = "6.4.0"
 django-tailwind = "^3.8.0"
 ruff = "^0.2.1"
+django = "^5.0.2"
+dockerflow = "^2024.2.0"
+django-filter = "^23.5"
```
Comment (yashikakhurana) [inline experimenter/pyproject.toml:76]: 🎉

**Review by yashikakhurana — APPROVED** [review]
yay lets go :shipit: , thanks  @jaredlockhart

File: experimenter/experimenter/openidc/middleware.py (hunk 1 above)
Comment (brennie) [reply experimenter/experimenter/openidc/middleware.py:34]: We should also probably fix that, thats jank as hell.

Comment (jaredlockhart) [reply experimenter/experimenter/openidc/middleware.py:34]: It is indeed.  Filed as a followup: https://github.com/mozilla/experimenter/issues/10318
